pyaedt/modules/solutions.py

```diff
@@ -1071,338 +1075,250 @@ def ifft_to_file(
+    def taper(self):
+        """Taper type.
+
+        Options are:
+        - ``"cosine"``
+        - ``"flat"``
+        - ``"hamming"``
+        - ``"triangular"``
+        - ``"uniform"``
         """
-        a = int(a)
-        b = int(b)
+        return self._taper
 
-        x = (a + 0.5) * self.Ax + (b + 0.5) * self.Bx
-        y = (a + 0.5) * self.Ay + (b + 0.5) * self.By
-        x_dis = x - self.CenterX
-        y_dis = y - self.CenterY
+    @taper.setter
+    def taper(self, val):
+        if val.lower() in ("flat", "uniform", "cosine", "triangular", "hamming"):
+            self._taper = val
+        else:
+            self.logger.error("This taper is not implemented")
 
-        return np.array([x_dis, y_dis, 0])
+    @property
+    def origin(self):
+        """Far field origin in meters."""
+        return self._origin
+
+    @origin.setter
+    def origin(self, vals):
+        if len(vals) != 3:
+            self.logger.error("Origin is wrong.")
+        else:
+            self._origin = vals
+            self.farfield_data = self.combine_farfield()
 
     @pyaedt_function_handler()
-    def assign_weight(self, a, b, taper="flat", port_cont=0):
+    def _assign_weight(self, a, b):
         """Assign weight to array.
 
         Parameters
         ----------
         a : int
-            Inndex of array, column.
+            Index of array, column.
         b : int
-            Inndex of array, row.
-        taper : string, optional
-            This is the type of taper we want to apply. The default is 'flat'.
-            It can be ``"cosine"``, ``"triangular"``, ``"hamming"`` or ``"flat"``.
+            Index of array, row.
 
         Returns
         -------
         float
-            Weight to applied to specific index of array.
+            Weight applied to specific index of the array.
         """
+        taper = self.taper
 
-        a = int(a)
-        b = int(b)
-        if taper.lower() == "flat":  # Flat
-            return self.mag_offset[port_cont]
+        if taper.lower() in ("flat", "uniform") or not self._is_array[self._freq_index]:
+            return 1
 
         cosinePow = 1
         edgeTaper_dB = -200
 
         edgeTaper = 10 ** ((float(edgeTaper_dB)) / 20)
 
         threshold = 1e-10
-        length_in_direction1 = 0
-        max_length_in_dir1 = 0
-        length_in_direction2 = 0
-        max_length_in_dir2 = 0
-        w1 = w2 = None
 
         # find the distance between current cell and array center in terms of index
-        length_in_direction1 = a - self.CenterA
-        length_in_direction2 = b - self.CenterB
-        max_length_in_dir1 = self.AMax
-        max_length_in_dir2 = self.BMax
+        lattice_vector = self._lattice_vector[self._freq_index]
+        if not lattice_vector or not len(lattice_vector) == 6:
+            return 1
+
+        center_a = (self.a_min + self.a_max) / 2
+        center_b = (self.b_min + self.b_max) / 2
+
+        length_in_direction1 = a - center_a
```

Comment:
Is "a" an index or a length? From this calculation it looks like a length, but it is indicated as index.